test/Makefile

```diff
@@ -0,0 +1,68 @@
+LIBDIR=../lib
+GTEST_MODULE=$(LIBDIR)/googletest
+GTEST_DIR=$(LIBDIR)/googletest/googletest
+GTEST_SOURCEDIR=$(GTEST_DIR)/src
+GTEST_HEADERS=$(GTEST_DIR)/include/gtest/*.h \
+              $(GTEST_DIR)/include/gtest/internal/*.h
+GTEST_SRCS_=$(GTEST_SOURCEDIR)/*.cc $(GTEST_SOURCEDIR)/*.h $(GTEST_HEADERS)
+
+# TODO: Factor out the common variables.
+CPP_NETLIB_DIR=$(LIBDIR)/cpp-netlib
+YAML_CPP_DIR=$(LIBDIR)/yaml-cpp
+
+CPPFLAGS+=-isystem $(GTEST_DIR)/include
+CXXFLAGS=-std=c++11 -g -pthread
+
+# Where to find code under test.
+SRC_DIR=../src
+
+TEST_DIR=.
+TEST_SOURCES=$(wildcard $(TEST_DIR)/*_unittest.cc)
+TEST_OBJS=$(TEST_SOURCES:%.cc=%.o)
```

Comment:
[Optional] I realized later that `$(TEST_DIR)` is `.`, so this is optional, but may be worth doing in case we end up putting tests in a subdirectory, lest we forget to override this (since the test executables do depend on the `.o` files being in the current directory).

Reply:
Given that we already have the $(TEST_DIR) variable, makes sense to use it explicitly. Done.